quixstreams/sources/kafka/kafka.py

```diff
@@ -0,0 +1,332 @@
+
+        if consumer_extra_config is None:
+            consumer_extra_config = {}
+
+        if consumer_group is None:
+            consumer_group = name
+
+        if auto_offset_reset is None:
+            auto_offset_reset = app_config.auto_offset_reset
+
+        self._config = app_config
+
+        self._topic = topic
+        self._on_consumer_error = on_consumer_error
+        self._consumer_poll_timeout = (
+            consumer_poll_timeout or self._config.consumer_poll_timeout
+        )
+        self._flush_timeout = consumer_extra_config.get(
+            "max.poll.interval.ms", self._config.flush_timeout
+        )
+        self._value_deserializer = value_deserializer
+        self._key_deserializer = key_deserializer
+
+        self._source_cluster_consumer = Consumer(
+            broker_address=broker_address,
+            consumer_group=consumer_group,
+            auto_offset_reset=auto_offset_reset,
+            auto_commit_enable=False,
+            extra_config=consumer_extra_config,
+        )
+        self._source_cluster_admin = TopicAdmin(
+            broker_address=broker_address,
+            extra_config=consumer_extra_config,
+        )
+
+        self._target_cluster_consumer = Consumer(
+            broker_address=self._config.broker_address,
+            consumer_group=f"{self._config.consumer_group}-offsets",
+            auto_offset_reset=self._config.auto_offset_reset,
+            auto_commit_enable=False,
+            extra_config=self._config.consumer_extra_config,
+        )
+        self._target_cluster_admin = TopicAdmin(
+            broker_address=self._config.broker_address,
+            extra_config=self._config.consumer_extra_config,
+        )
+
+        self._running = True
+        self._error: Optional[Exception] = None
+        self._checkpoint: Optional[Checkpoint] = None
+
+    def run(self) -> None:
+        self._validate_topics()
+
+        self._source_cluster_consumer.subscribe(
+            topics=[self._topic],
+            on_assign=self.on_assign,
+            on_lost=self.on_lost,
+            on_revoke=self.on_revoke,
+        )
+
+        super().run()
+
+        self.init_checkpoint()
+        try:
+            while self._running:
+                if self._error:
+                    raise self._error
+
+                msg = self.poll_source()
+                if msg is None:
+                    continue
+
+                self.produce_message(msg)
+                self.commit_checkpoint()
+        except Exception:
+            self._checkpoint.close()
+            raise
+
+        self.commit_checkpoint(force=True)
+
+    def produce_message(self, msg: Message):
+        topic_name, partition, offset = msg.topic(), msg.partition(), msg.offset()
+        self._checkpoint.store_offset(topic_name, partition, offset)
+        logger.log(
+            5,
+            "producing message to partition %s with offset %s",
+            partition,
+            offset,
+        )
```

Comment:
Why to use `logger.log(5...)` here instead of `logger.debug` or similar?

Reply:
Because it's very very spamy but useful when doing in depth debugging of the source. 5 is lower than debug.

Reply:
In that case, how is it going to be visible if most people use `DEBUG`, `INFO` and similar values to set a log level?

Reply:
It's not. That's the point. We don't want to have one log per message in `DEBUG`. I'd rather remove the log if that's an issue.

Reply:
If there's no easy way to print this log, then let's remove it 👍